mobx/lib/src/api/observable_collections/observable_map.dart

```diff
@@ -60,12 +60,11 @@ class ObservableMap<K, V>
   @override
   void operator []=(K key, V value) {
     _context.conditionallyRunInAction(() {
-      V oldValue;
+      final oldValue = _map[key];
```

Comment:
Good find!

Reply:
I think you should start making branches on mobx.dart, so we can get the CI tests to detect any potential failures. Can you please submit the PR from a mobx.dart branch? I've sent you the member invite already ;-)

Reply:
Thanks!
Sure.